src/jsifier.js

```diff
@@ -494,7 +494,7 @@ function JSify(data, functionsOnly) {
       print(preprocess(read('arrayUtils.js')));
     }
 
-    if (SUPPORT_BASE64_EMBEDDING) {
+    if (SUPPORT_BASE64_EMBEDDING && !MINIMAL_RUNTIME) {
```

Comment:
why not have minimal runtime use the base64Utils?

Reply:
(Do the two runtimes use different base64 logic? If so why?)

Reply:
Yes, the two runtimes have different base64 code. The reason for that is to not break users. The base64utils.js code is exposed publicly to users. If it had been an internal one, I would have removed it, but it is a public one, and has different semantics of throwing an error on incorrect base64 input, instead of expecting always valid base 64 input.

Reply:
I think the base64 code was an internal API, and it's fine if we change things. Unless we documented the API publicly and implied it would be stable?

I think the semantics change is good too, we should only be reading our own base64 code.

Unifying this would make things significantly simpler which I think is worth it, for low risk of changing an internal API.

Reply:
Are you sure?

https://github.com/emscripten-core/emscripten/blame/master/src/modules.js#L465 has an explicit export?

https://github.com/emscripten-core/emscripten/pull/6007

Reply:
Oh, good point, if it's allowed to be exported then let's not change it.